### DISCUSSION TITLE
Unread event handler

### DISCUSSION THREAD

**Discussion by bsbarkur** [chatwoot/chatwoot#2185] [Q&A]
So where in the chatwoot codebase, unread message event handler has been implemented?
What is the purpose of this function `def unread_messages` in api/v1/accounts/conversations_controller.rb`

**To Reproduce**

1. Client connects to chatwoot agent.
2. Client disconnects from chatwoot agent and the conversation is not resolved.
3. Agent sends message reply which is unread by the client as it is not connected to chatwoot.
4. So what unread event controller exists in place at the client side ?

**Expected behavior**
Trigger unread message event handler to push a message back to the client. Useful for push notifications back to the client.

**Environment**
Self hosted chatwoot instance on openshift cloud.

**Comment by pranavrajs** [comment]
@bsbarkur This seems like a question. Moving it to discussions. Please feel free to reopen it is an error. 

On the actual issue, I don't understand what you are trying to solve. If you could explain the context a little bit more, it would be helpful.

**Reply by bsbarkur** [reply]
thanks for moving to the appropriate place. so if I want to check if connected client is offline, how do I do that ? I know how to detect the chatwoot user availability status. But can we do the same for the client when client has disconnected from chatwoot server ?

**Reply by pranavrajs** [reply]
Contact also has the availability status attribute same as the user. I guess you would be able to make use of it.

**Comment by bsbarkur** [comment]
Was able to make use of this. Working well. Thanks.
